src/widget/form/profileform.cpp

```diff
@@ -416,3 +431,83 @@ void ProfileForm::retranslateUi()
     // We have to add the toxId tooltip here and not in the .ui or Qt won't know how to translate it dynamically
     toxId->setToolTip(tr("This bunch of characters tells other Tox clients how to contact you.\nShare it with your friends to communicate."));
 }
+
+void ProfileForm::showRegisterToxme()
+{
+    bodyUI->toxmeUsername->setText("");
+    bodyUI->toxmeBio->setText("");
+    bodyUI->toxmePrivacy->setChecked(false);
+
+    bodyUI->toxmeRegisterButton->show();
+    bodyUI->toxmeUpdateButton->hide();
+    bodyUI->toxmePassword->hide();
+    bodyUI->toxmePasswordLabel->hide();
+}
+
+void ProfileForm::showExistenToxme()
+{
+    QStringList info = Settings::getInstance().getToxmeInfo().split("@");
+    bodyUI->toxmeUsername->setText(info[0]);
+    bodyUI->toxmeServersList->addItem(info[1]);
+
+    QString bio = Settings::getInstance().getToxmeBio();
+    bodyUI->toxmeBio->setText(bio);
+
+    bool priv = Settings::getInstance().getToxmePriv();
+    bodyUI->toxmePrivacy->setChecked(priv);
+
+    QString pass = Settings::getInstance().getToxmePass();
+    bodyUI->toxmePassword->setText(pass);
+    bodyUI->toxmePassword->show();
+    bodyUI->toxmePasswordLabel->show();
+
+    bodyUI->toxmeRegisterButton->hide();
+    bodyUI->toxmeUpdateButton->show();
+}
+
+void ProfileForm::onRegisterButtonClicked()
+{
+    QString name =  bodyUI->toxmeUsername->text();
+    if (name.isEmpty())
+        return;
+
+    bodyUI->toxmeRegisterButton->setEnabled(false);
+    bodyUI->toxmeUpdateButton->setEnabled(false);
+    bodyUI->toxmeRegisterButton->setText(tr("Register (processing)"));
+    bodyUI->toxmeUpdateButton->setText(tr("Update (processing)"));
+
+    QString id = toxId->text();
+    QString bio = bodyUI->toxmeBio->text();
+    QString server = bodyUI->toxmeServersList->currentText();
+    bool privacy = bodyUI->toxmePrivacy->isChecked();
+
+    Core* oldCore = Core::getInstance();
+
+    Toxme::ExecCode code = Toxme::ExecCode::Ok;
+    QString response = Toxme::createAddress(code, server, id, name, privacy, bio);
+
+    Core* newCore = Core::getInstance();
+    if (oldCore == newCore)
```

Comment:
This about "oldCore == newCore"?
It's a long-long story started in #2753 :)
If user will click register and quick logout, it will have segfault. It fixed by check newCore != nullptr.
But if user will quickly logout and login, it still have error. And added compare oldCore and newCore for check both